integrationtests/src/test/java/io/kroxylicious/proxy/multitenant/MultiTenantIT.java

```diff
@@ -187,6 +188,50 @@ public void tenantIsolation(KafkaCluster cluster) throws Exception {
         }
     }
 
+    @Test
+    public void tenantGroupIsolation(KafkaCluster cluster) throws Exception {
+        String config = getConfig(PROXY_ADDRESS, cluster);
+        try (var proxy = startProxy(config)) {
+            // tenant 1 - creates topic
+            createTopics(TENANT1_PROXY_ADDRESS, List.of(NEW_TOPIC_1));
+            // tenant 1 - creates a consumer
+            createConsumerWithGroup(TENANT1_PROXY_ADDRESS, "Tenant1Group", NEW_TOPIC_1);
+            // tenant 1 - uses kafka admin api to check consumer groups
+            verifyConsumerGroups(TENANT1_PROXY_ADDRESS, "Tenant1Group");
+
+            // tenant 2 - creates a topic
+            createTopics(TENANT2_PROXY_ADDRESS, List.of(NEW_TOPIC_1));
+            // tenant 2 - creates a consumer
+            createConsumerWithGroup(TENANT2_PROXY_ADDRESS, "Tenant2Group", NEW_TOPIC_1);
+            // tenant 2 - uses kafka admin api to check consumer groups, and it will fail because it can see tenant 1 also
+            verifyConsumerGroups(TENANT2_PROXY_ADDRESS, "Tenant2Group");
+        }
+    }
+
+    private void createConsumerWithGroup(String proxyAddress, String groupId, NewTopic topic) {
```

Comment:
I anticipate that we'll refactor these methods more and we build up the for consumer use-cases.

Reply:
Yeah I was musing on that after commenting on [PR#201](https://github.com/kroxylicious/kroxylicious/pull/201#discussion_r1141683854).

In that comment thread I was thinking of including something in the junit-extension. However I realise that would give us a circular dependency. So we probably need a second extension. The current one provides manages Kafka Clusters and thus can be depended on by Kroxylicious for tests. However if we want an extension to manage Proxies and proxy clients we need something different as it needs to be able to depend on Kroxylicious. 

It would be very useful if the extension(s) had some way of identifying [virtual|physical] clusters and determining their bootstrap URL's for use when creating a client. Then we could do something like 
```java
public void shouldProduceMessage(@VirtualCluster(clusterId="tennant1") KafkaProducer virtualProducer, @PhysicalCluster(clusterId="uberCluster") KafkaConsumer physicalConsumer) {
    virtualProducer.send(X);

    var records = physicalConsumer.poll();
    assertThat(records).containsExactly(Y);
}
``` 

For the record  I'm not expecting this PR to address this thought, I'm just voicing it now as its the second PR 24 hours looking at the same problem.

Reply:
Agreed, something to look into later!